src/dune_console/dune_console.ml

```diff
@@ -131,26 +143,66 @@ module Backend = struct
 
   let spawn_thread = Fdecl.create Dyn.opaque
 
-  let threaded (module Base : S) : (module S) =
+  (** Backends may have access to an internal state. This type is used by the
+      [Threaded] backends. *)
+  type state =
+    { messages : User_message.t Queue.t
+    ; mutable finish_requested : bool
+    ; mutable finished : bool
+    ; mutable status_line : User_message.Style.t Pp.t option
+    }
+
+  (** [Threaded] is the interface for user interfaces that are rendered in a
+      separate thread. It can be used to directly construct a threaded backend. *)
+  module type Threaded = sig
```

Comment:
This is module interface that will be used by both the current console ui and the upcoming tui.

Notice that the tui can also react to keyboard input, so it has the additional `handle_user_events` call.

Reply:
Why not add this comment to the code?

Reply:
What exactly should we add to the code? We already document the point of `Threaded` and `handle_user_events`.

Reply:
My understanding is that right now `handle_user_events` is just unused and only you, Rudi and whoever happened to read this comment here knows why it exists. Well, this "upcoming tui" might never happen for a variety of reasons, so why not leave the context in the code too?

Reply:
(Adding Rudi's comment verbatim is just fine!)